### PR TITLE
Creation of model for code snippet question

### PR DESCRIPTION
**Pull request by Deepankar-promact**
Task#: 20889
File changes:
CodeSnippetQuestion.cs : created model for code snippet question
CodingLanguage.cs : created model for coding language
QuestionLanguageMapping.cs : created model for mapping  code snippet question & coding language
CodeSnippetQuestionModel.cs : created model for storing question data received from client side
SeedDatabase.cs : created to seed predefined coding language into database on runtime.

### CODE REVIEW

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/ProgramingLanguageEnum.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Promact.Trappist.DomainModel.ApplicationClass
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/ProgramingLanguageEnum.cs]: enum should not be in within application class folder

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            context.Database.EnsureCreated();
+
+            if (context.CodingLanguage.Any())
+            {
+                return;
+            }
+
+            var Languages = new CodingLanguage[]
+            {
+                new CodingLanguage {Language = ProgramingLanguageEnum.java,  CreatedDateTime = DateTime.UtcNow, UpdateDateTime = DateTime.UtcNow },
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs]: please remove createddatetime and updatedateime assignments. it has been handled in dbcontext.

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    public class SeedDatabase
+    {
+        public static object ProgramingLanguage { get; private set; }
+
+        /// <summary>
+        /// Seed data into the table
+        /// </summary>
+        /// <param name="context">DbContext</param>
+        public static void Seed(TrappistDbContext context) {
+            context.Database.EnsureCreated();
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs]: no need to check database is created or not.

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodeSnippetQuestion.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace Promact.Trappist.DomainModel.Models.Question
+{
+    //To-Do inherit QuestionBaseModel instead of BaseModel once it is created
+    public class CodeSnippetQuestion : BaseModel
+    {
+
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodeSnippetQuestion.cs]: please remove extra space

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodeSnippetQuestion.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        [Required]
+        public bool RunBasicTestCase { get; set; }
+
+        [Required]
+        public bool RunCornerTestCase { get; set; }
+
+        [Required]
+        public bool RunNecessaryTestCase { get; set; }
+
+        public virtual QuestionLanguageMapping QuestionLanguangeMapping { get; set; }
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodeSnippetQuestion.cs]: how many to many relationships established here?

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/ProgramingLanguageEnum.cs (hunk 1 above)
Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/ProgramingLanguageEnum.cs]: Where should I put it?

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs (hunk 3 above)
Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs]: Ok.

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs (hunk 2 above)
Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs]: I will remove it.

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodeSnippetQuestion.cs (hunk 4 above)
Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/CodeSnippetQuestion.cs]: Ok

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionLanguageMapping.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    public class QuestionLanguageMapping: BaseModel
+    {
+        [Required]
+        public int QuestionId { get; set; }
+
+        [Required]
+        public int LanguageId { get; set; }
+
+        [ForeignKey("QuestionId")]
+        public CodeSnippetQuestion CodeSnippetQuestion { get; set; }
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionLanguageMapping.cs]: virtual keyword is missing.

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionLanguageMapping.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+        [ForeignKey("LanguageId")]
+        public CodingLanguage CodeLanguage { get; set; }
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/QuestionLanguageMapping.cs]: virtual keyword is missing

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs [hunk 8]
```diff
@@ -0,0 +1,42 @@
+﻿using Promact.Trappist.DomainModel.DbContext;
+using Promact.Trappist.DomainModel.Enum;
+using Promact.Trappist.DomainModel.Models.Question;
+using System.Linq;
+
+namespace Promact.Trappist.DomainModel.Data
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs]: namespace is wrong. Please update it.

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+{
+
+    /// <summary>
+    /// Polpulates pre-required table
+    /// </summary>
+    public class SeedDatabase
+    {
+        public static object ProgramingLanguage { get; private set; }
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs]: please remove unused property

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+using Promact.Trappist.DomainModel.Models.Question;
+using System.Linq;
+
+namespace Promact.Trappist.DomainModel.ApplicationClasses.Question
+{
+
+    /// <summary>
+    /// Polpulates pre-required table
+    /// </summary>
+    public class SeedDatabase
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/Question/SeedDatabase.cs]: seed database should be in domain model namespace. Create one folder and add your seed in that folder.

File: Trappist/src/Promact.Trappist.Web/Startup.cs [hunk 11]
```diff
@@ -108,6 +109,9 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
                      defaults: new { controller = "Home", action = "Index" });
             });
 
+            SeedDatabase seedDb = new SeedDatabase();
+            seedDb.Seed(context);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/Startup.cs]: please make seed method as an extension method.

File: Trappist/src/Promact.Trappist.DomainModel/Seed/SeedDatabase.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+using Promact.Trappist.DomainModel.Enum;
+using Promact.Trappist.DomainModel.Models.Question;
+using System.Linq;
+
+namespace Promact.Trappist.DomainModel.Seed
+{ 
+    /// <summary>
+    /// Polpulates pre-required table
+    /// </summary>
+    public static class SeedDatabase
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Seed/SeedDatabase.cs]: please add extensions word after class name which contains extensions method

Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.DomainModel/Seed/SeedDatabase.cs]: Ok

**Pull request closed without merging**